integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/MatchSearchPredicateIT.java

```diff
@@ -979,22 +979,32 @@ public void analyzerOverride_normalizedStringField() {
 	}
 
 	@Test
-	@TestForIssue( jiraKey = "HSEARCH-2534" )
+	@TestForIssue( jiraKey = { "HSEARCH-2534", "HSEARCH-3042" } )
 	public void analyzerOverride_queryOnlyAnalyzer() {
 		StubMappingScope scope = indexManager.createScope();
 		String absoluteFieldPath = indexMapping.whitespaceLowercaseAnalyzedField.relativeFieldName;
+		String ngramFieldPath = indexMapping.ngramSearchAnalyzedField.relativeFieldName;
 
+		// Using the white space lower case analyzer, we don't have any matching.
 		SearchQuery<DocumentReference> query = scope.query()
 				.predicate( f -> f.match().field( absoluteFieldPath ).matching( "worldofwordcraft" ) )
 				.toQuery();
-
 		assertThat( query ).hasNoHits();
 
+		// Overriding the analyzer with a n-gram analyzer for the specific query,
+		// the same query matches all values.
 		query = scope.query()
 				.predicate( f -> f.match().field( absoluteFieldPath ).matching( "worldofwordcraft" )
 						.analyzer( OverrideAnalysisDefinitions.ANALYZER_NGRAM.name ) )
 				.toQuery();
+		assertThat( query )
+				.hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1, DOCUMENT_2, DOCUMENT_3 );
 
+		// Defining a ngram search analyzer to override the analyzer,
+		// we expect the same result.
+		query = scope.query()
+				.predicate( f -> f.match().field( ngramFieldPath ).matching( "worldofwordcraft" ) )
+				.toQuery();
```

Comment:
Great idea! This is indeed very relevant.

Reply:
Thanks!